CONTRIBUTING.md

```diff
@@ -251,6 +251,20 @@ To keep naming patterns consistent across the project, naming patterns are enfor
   - `func CreateTracesExport(...) {...}`
   - `func CreateTracesToTracesFunc(...) {...}`
 
+### Enumerations
+
+To keep naming patterns consistent across the project, enumeration patterns are enforced to make intent clear:
+
+- Enumerations should be defined using a type alias
+- Enumerations should use either `int` or `string` as the underlying type
+- Enumeration name should be:
+  - Prefixed with a clarifying name *unless* the package name describes the entity of the enumeration.
+  - Suffixed with `Type`
```

Comment:
Starting with a strict statement here so we can poke holes at this.

Immediately an enumeration like `component.Kind`, which has constants like `ReceiverKind`, would not match this role.

Reply:
Why is `Type` as a literal required? Enums should be prefixed with _a type_. E.g. `configtelemetry.LevelNone` when we have `type Level int32`

Reply:
> An enum const should be prefixed with its type.

Agreed, I'll update that rule. 

I think the largest question of the PR is "When should an Enum use `Type` in its name or be named `Type`". I've started the PR by claiming "All Enums should include `Type` in the name".

I don't think that is correct, but I am struggling to put into words the situations when `Type` is appropriate.

Reply:
> I don't think that is correct, but I am struggling to put into words the situations when Type is appropriate.

I don't think we should dictate using `Type` as literal. Sometimes it may be a good fit, sometimes not

Reply:
@dmitryax Updated to try and put into words what `Type` represents, which I am currently interpreting as a sense of limited categorization.